fcrepo-clients/src/test/java/edu/unc/lib/dl/fcrepo4/BinaryObjectTest.java

```diff
@@ -0,0 +1,143 @@
+package edu.unc.lib.dl.fcrepo4;
+
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import static org.mockito.Matchers.eq;
+
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayInputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.List;
+
+import com.hp.hpl.jena.rdf.model.Model;
+import com.hp.hpl.jena.rdf.model.ModelFactory;
+import com.hp.hpl.jena.rdf.model.Resource;
+import edu.unc.lib.dl.fedora.ObjectTypeMismatchException;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.rdf.Ebucore;
+import edu.unc.lib.dl.rdf.Fcrepo4Repository;
+import edu.unc.lib.dl.rdf.Premis;
+
+public class BinaryObjectTest extends AbstractFedoraTest {
+	@Mock
+	private PID mockPid;
+	
+	private BinaryObject binObj;
+	private Model model;
+	private Resource resource;
+	
+	private ByteArrayInputStream stream;
+	
+	private static final String BASE_PATH = "http://www.myrepo.com";
+	private static final String METADATA_PATH = "/fcr:metadata";
```

Comment:
You may also want to use RepositoryPathConstants.FCR_METADATA instead of making a new constant